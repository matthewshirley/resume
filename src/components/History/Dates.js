import React from 'react';
import PropTypes from 'prop-types';

import Text from '../common/Text';

const Dates = ({ startDate, endDate }) => {
  if (!startDate && !endDate) return (null);

  const isPresent = !endDate || endDate === 'Jan 1900'; // TODO: Multiple GraphQL types?
  const formattedEndDate = isPresent ? 'Present' : endDate;

  return (
    <Text m={0}>
      {`${startDate} - ${formattedEndDate}`}
    </Text>
  );
};

Dates.propTypes = {
  startDate: PropTypes.string,
  endDate: PropTypes.string,
};

Dates.defaultProps = {
  startDate: null,
  endDate: null,
};

export default Dates;
