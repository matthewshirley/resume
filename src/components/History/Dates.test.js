import React from 'react';
import renderer from 'react-test-renderer';

import Dates from './Dates';


describe('Dates', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Dates startDate="Jan 2017" endDate="Dec 2017" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with the end date as present', () => {
    const tree = renderer
      .create(<Dates startDate="Jan 2017" endDate={null} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('does not render without a start and end date', () => {
    const tree = renderer
      .create(<Dates startDate={null} endDate={null} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
