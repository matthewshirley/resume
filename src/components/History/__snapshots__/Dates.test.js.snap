// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary does not render without a start and end date 1`] = `null`;

exports[`Summary renders correctly 1`] = `
<p
  className="sc-AxjAm bnsefY"
  fontSize={1}
>
  Jan 2017 - Dec 2017
</p>
`;

exports[`Summary renders with the end date as present 1`] = `
<p
  className="sc-AxjAm bnsefY"
  fontSize={1}
>
  Jan 2017 - Present
</p>
`;
